### ISSUE TITLE
When the name is empty and no PropertySourceLoader match, the resource load should be interrupted

### ISSUE THREAD

**Issue by brucelwl** [spring-projects/spring-boot#19184]
class ``org.springframework.boot.context.config.ConfigFileApplicationListener``

method ``private void load(String location, String name, Profile profile, DocumentFilterFactory filterFactory,DocumentConsumer consumer)``

When the name is empty and no PropertySourceLoader match, the resource load should be interrupted

![image](https://user-images.githubusercontent.com/13413715/69839848-3a40b380-1294-11ea-8fc5-9cb0268a908d.png)

The code should be changed as follows:

```
private void load(String location, String name, Profile profile, 
           DocumentFilterFactory filterFactory,DocumentConsumer consumer) {
	if (!StringUtils.hasText(name)) {
		for (PropertySourceLoader loader : this.propertySourceLoaders) {
			if (canLoadFileExtension(loader, location)) {
				load(loader, location, profile, filterFactory.getDocumentFilter(profile), consumer);
					return;
			}
		}
		return; //add break
	}
	Set<String> processed = new HashSet<>();
	for (PropertySourceLoader loader : this.propertySourceLoaders) {
	     for (String fileExtension : loader.getFileExtensions()) {
	        if (processed.add(fileExtension)) {
	          loadForFileExtension(loader, location + name, "." + fileExtension, profile, filterFactory,consumer);
		   }
		}
	}
}
```

**Comment by spencergibb** [comment]
This is not the right repo for spring cloud commons

**Comment by brucelwl** [comment]
> This is not the right repo for spring cloud commons

You are right, But I submitted the issue in springboot repo ! ! !

**Comment by wilkinsona** [comment]
Thanks for the suggestion, @brucelwl. In the case where `name` is an empty string, the call to `load` at the end of `loadForExtension` is redundant, but there's quite a lot of other logic earlier in the same method. On first look, it's not clear to me that it can be skipped. To avoid duplication of effort, can you please share your analysis that led to you suggesting the early return?

**Comment by brucelwl** [comment]
If I set a file directly through ``-Dspring.config.additional-location=classpath:application.json``, 
instead of a folder, the parameter ``name`` will be empty, but if there is no suitable 
``PropertySourceLoader`` at this time, the for loop will not be interrupted. 

 If it is executed downwards, the name of file will be concatenated with ``name``,and the result is ``application.jsonnull``, which is a non-existent file name. 

If there is no matching a, it is not necessary to continue execution, and you should add ``break``

**Comment by philwebb** [comment]
The fix for #17241 (commit 0f105f747124f5ea349b7148f8e8c3e22d0865f6) throws an `IllegalStateException` instead of using `return`. I think we decided against back-porting that fix to 2.1.x  just in case there were unexpected side effects.

I think we can consider this one a duplicate of #17241
